content/blog/2022-01-15-moving-local-experiments-to-the-cloud-with-iterative-terraform-provider.md

```diff
@@ -0,0 +1,280 @@
+- _script_ (**required)**: This is where the Iterative Provider's magic happens,
+  i.e. this is where we define a script that should be run on a provisioned
+  machine.
+
+Tke a look at the script. In the simplest scenario, all we need to do on a new
+machine to run the training script is to set up the Python environment with
+required libraries, and run the script. And that's exactly what the script here
+does (granted you have a `requirements.txt` file). If you simply want to train
+your model on a machine with more memory, this may be enough. However, if you
+want your training code to leverage GPUs, the script will look a bit more
+complex.
+
+## Training with GPU
+
+Setting up the environment to leverage GPU devices is a bit involved as you need
+to make sure you have the required drivers, NVIDIA-docker, and CUDA.
+
+```
+resource "iterative_task" "tpi-examples-gpu" {
+    name      = "tpi-examples-gpu"
+    cloud     = "aws"
+    region    = "us-east-2"
+    machine   = "l+k80"
+    directory = "."
+
+    script = <<-END
+    #!/bin/bash
+    sudo apt update
+    sudo apt-get install -y software-properties-common build-essential ubuntu-drivers-common
+    sudo ubuntu-drivers autoinstall
+
+    sudo curl -fsSL https://get.docker.com -o get-docker.sh && sudo sh get-docker.sh &&
+    sudo usermod -aG docker ubuntu
+    sudo setfacl --modify user:ubuntu:rw /var/run/docker.sock
+    curl -s -L https://nvidia.github.io/nvidia-docker/gpgkey | sudo apt-key add -
+    curl -s -L https://nvidia.github.io/nvidia-docker/ubuntu20.04/nvidia-docker.list | sudo tee /etc/apt/sources.list.d/nvidia-docker.list
+    sudo apt update && sudo apt install -y nvidia-docker2
+    sudo systemctl restart docker
+    rm get-docker.sh
+
+    nvidia-smi
+    docker run --rm --gpus all -v "$PWD:/tpi" iterativeai/cml:0-dvc2-base1-gpu \
+        /bin/bash -c "cd /tpi; pip install -r requirements.txt; python src/train.py"
+    END
+}
+```
+
+Here, to run my training code I am passing my project's directory as a volume to
+a Docker image provided by Iterative (`iterativeai/cml:0-dvc2-base1-gpu`) which
+takes care of CUDA.
+
+## Ready… Set… Apply!
+
+Whether you want to go with the basic example, or the GPU-enabled training, you
+can run `terraform plan` to review what steps Terraform is going to take to make
+your desired infrastructure materialize. Don't worry, nothing is actually done
+at this point, but it's a good way to check for potential issues in the
+configuration.
+
+If there are no issues, run `terraform apply` to execute the actions proposed in
+the Terraform plan. You'll need to type `yes` to confirm that you indeed would
+like the plan to be executed.
+
+At this point you can get a cup of your favourite beverage, and let the
+Iterative Provider work its magic together with Terraform. They will allocate a
+remote machine for you, set up Docker, pull required images, install missing
+drivers, and finally run your `train.py`. Once the script finishes, the machine
```

Comment:
```suggestion
remote machine for you, install missing drivers, set up dependencies,
and finally run your `train.py`. Once the script finishes, the machine
```

Reply:
Resolved in b0443d1e629d77d7b44cf9b3c9f444f80629913a